MapleServer2/Types/Stat.cs

```diff
@@ -8,47 +9,83 @@ namespace MapleServer2.Types;
  *
  * Change Stat.Bonus for temporary changes.
  */
-public class Stat
+public struct StatValue
 {
     public long BonusLong;
     public long BaseLong;
     public long TotalLong;
-    public int Bonus { get => (int) BonusLong; protected set { BonusLong = value; } }
-    public int Base { get => (int) BaseLong; protected set { BaseLong = value; } }
-    public int Total { get => (int) TotalLong; protected set { TotalLong = value; } }
+    public int Bonus { get => (int) BonusLong; set { BonusLong = value; } }
+    public int Base { get => (int) BaseLong; set { BaseLong = value; } }
+    public int Total { get => (int) TotalLong; set { TotalLong = value; } }
+    public long BonusAmountLong { get => BonusLong - BaseLong; }
+    public int BonusAmount { get => (int) BonusAmountLong; }
```

Comment:
What is bonus amount, isn't bonus already "bonus"??

Reply:
BonusAmount is used to get the bonus value that would be displayed on the advanced stats page. It's used in places like p/m attack calculations since that happens in two parts.